oozie-to-airflow/mappers/fs_mapper.py

```diff
@@ -0,0 +1,198 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Maps FS node to Airflow's DAG"""
+
+import shlex
+from typing import Set, NamedTuple, List
+from xml.etree.ElementTree import Element
+
+from converter.parser import Relation
+from mappers.action_mapper import ActionMapper
+from utils.template_utils import render_template
+from utils.el_utils import normalize_path
+from utils.xml_utils import bool_value
+
+ACTION_TYPE = "fs"
+
+FS_OP_MKDIR = "mkdir"
+FS_OP_DELETE = "delete"
+FS_OP_MOVE = "move"
+FS_OP_CHMOD = "chmod"
+FS_OP_TOUCHZ = "touchz"
+FS_OP_CHGRP = "chgrp"
+FS_OP_SETREP = "setrep"
+
+FS_TAG_PATH = "path"
+FS_TAG_SOURCE = "source"
+FS_TAG_TARGET = "target"
+FS_TAG_RECURSIVE = "recursive"
+FS_TAG_DIRFILES = "dir-files"
+FS_TAG_SKIPTRASH = "skip-trash"
+FS_TAG_PERMISSIONS = "permissions"
+FS_TAG_GROUP = "group"
+FS_TAG_REPLFAC = "replication-factor"
+
+
+def prepare_mkdir_command(node: Element, params):
+    path = normalize_path(node.attrib[FS_TAG_PATH], params)
+    command = "fs -mkdir {path}".format(path=shlex.quote(path))
+    return command
+
+
+def prepare_delete_command(node: Element, params):
+    path = normalize_path(node.attrib[FS_TAG_PATH], params)
+    skip_trash = bool_value(node, FS_TAG_SKIPTRASH)
+    extra = " -skipTrash" if skip_trash else ""
+    command = "fs -rm -r {extra} {path}".format(extra=extra, path=shlex.quote(path))
+
+    return command
+
+
+def prepare_move_command(node: Element, params):
+    source = normalize_path(node.attrib[FS_TAG_SOURCE], params)
+    target = normalize_path(node.attrib[FS_TAG_TARGET], params)
+
+    command = "fs -mv {source} {target}".format(source=source, target=target)
+    return command
+
+
+def prepare_chmod_command(node: Element, params):
+    path = normalize_path(node.attrib[FS_TAG_PATH], params)
+    permission = node.attrib[FS_TAG_PERMISSIONS]
+    # TODO: Add support for dirFiles
```

Comment:
I think it would be best not to leave TODOs in the code, but create issues in Github instead.
We agreed on such an approach with @potiuk some time ago.

Reply:
Maybe add #80 to the TODO :). By the way - it's great with all the github integration that you can start typing "#" in the comment and text and it will hint you the opened issues - without leaving the comment field :).

Reply:
I updated a PR.